### ISSUE TITLE
Plugin conflict during vim-fugitives :Gdiff command

### ISSUE THREAD

**Issue by lervag** [neoclide/coc.nvim#1023]
## Describe the bug

When saving the fugitive Git index buffer in `:Gdiff`, the following error is displayed by coc.nvim:

```
[coc.nvim] error: Uncaught exception: Error: ENOENT: no such file or directory, open '/home/user/test/.git//0/test.txt'
```

## Reproduce the bug

1. Preparation

    ```sh
    mkdir test
    cd test
    git init
    echo "hello world" > test.txt
    git add test.txt
    git ci -m "..."
    git clone -b release https://github.com/neoclide/coc.nvim
    git clone https://github.com/tpope/vim-fugitive
    ```

2. Create minimal vimrc file as below

    ```vim
    set nocompatible
    let &runtimepath = 'vim-fugitive,' . &runtimepath
    let &runtimepath = 'coc.nvim,' . &runtimepath
    filetype plugin indent on
    syntax enable

    let g:coc_extension_root = $HOME . `/test/coc-config'
    let g:coc_global_extensions = ['coc-snippets']

    nnoremap q :qall!<cr>
    ```

3. `nvim -u test.vim test.txt`

    - Delete "Hello" and `:write`
    - Do `:Gdiff`
    - Delete "Hello" from the fugitive buffer and `:write`

After the last step, observe the reported error. The expected result is for there to be no error.

## Result from CocInfo

```
## versions

vim version: NVIM v0.4.0-1285-g519382646
node version: v11.15.0
coc.nvim version: 0.0.73-800fdeacc6
term: tmux-256color-italic
platform: linux

## Messages

## Output channel: snippets

```

**Comment by chemzqm** [comment]
Please provide the log from `:CocOpenLog`

**Comment by lervag** [comment]
Output is:

```
2019-07-25T23:24:20.016 INFO (pid:26489) [plugin] - coc 0.0.73-800fdeacc6 initialized with node: v11.15.0
2019-07-25T23:24:32.192 INFO (pid:26489) [willSaveHandler] - Will save cost: 2
2019-07-25T23:24:32.219 ERROR (pid:26489) [server] - uncaughtException Error: ENOENT: no such file or directory, open '/home/lervag/test/.git//0/test.txt'
```

**Comment by lervag** [comment]
Notice that my `coc-settings.json` is also empty when performing this test.

**Comment by chemzqm** [comment]
Can't reproduce, try add `let $COC_NO_PLUGINS=1` to the top of vimrc to disable extensions.

**Comment by lervag** [comment]
Yes, that removes the error. That is strange. How can I see which plugins are loaded?

**Comment by lervag** [comment]
Ah, with `:CocList extensions` I see that the following extensions were loaded:

```
    1 * coc-ultisnips 1.2.3 ~/.config/coc/extensions/node_modules/coc-ultisnips
    2 * coc-snippets 2.1.8 ~/.config/coc/extensions/node_modules/coc-snippets
    3 * coc-omni 1.2.4 ~/.config/coc/extensions/node_modules/coc-omni
    4 + coc-yaml 1.0.2 ~/.config/coc/extensions/node_modules/coc-yaml
    5 + coc-vimtex 1.0.2 ~/.config/coc/extensions/node_modules/coc-vimtex
    6 + coc-python 1.2.4 ~/.config/coc/extensions/node_modules/coc-python
    7 + coc-json 1.2.3 ~/.config/coc/extensions/node_modules/coc-json
```

**Comment by chemzqm** [comment]
Try disable extensions in `:CocList exntensions`

**Comment by lervag** [comment]
Ok, the issue is due to the `coc-snippets` extension.

**Comment by lervag** [comment]
I'm curious, is it possible to easily specify different directories for the `coc-settings.json` and the `~/.config/coc` files? That would make it easier to make a clean minimal example. I've now several times realized I needed to temporarily remove files in order to make the example become clean.

**Comment by chemzqm** [comment]
Use `g:coc_extension_root` for custom extension root.

**Comment by chemzqm** [comment]
I have `coc-snippets` but can't reproduce the issue, and it should be stack trace in the log

**Comment by lervag** [comment]
I don't find any stack traces, sorry. I've updated my OP with instructions that reproduce this for me.

**Comment by lervag** [comment]
I'm more than happy to help debug this, but I need some help. The `index.js` file in coc-snippets is ~10000 lines long, and the `CocOpenLog` does not show me any stack trace. Thus it is very hard to investigate where the exception is raised.

If you have some pointers on how I can try to pinpoint the place it occurs, then that would be very helpful.

**Comment by chemzqm** [comment]
Fixed on coc-snippets@2.1.9

**Comment by lervag** [comment]
Great, thanks! And thanks for the impressive work on this plugin its extensions!
